engine/src/main/java/org/hibernate/search/bridge/util/impl/NumericFieldUtils.java

```diff
@@ -118,9 +118,12 @@ public static boolean requiresNumericRangeQuery(Object value) {
 			return false;
 		}
 		return value instanceof Double ||
+				value instanceof Byte ||
+				value instanceof Short ||
 				value instanceof Long ||
 				value instanceof Integer ||
 				value instanceof Float ||
+				value instanceof Date ||
```

Comment:
Not really. This method is only used with Infinispan (when I did it, I wondered how it could have already been working with Es). See `ConnectedMultiFieldsRangeQueryBuilder` and the comment "//we need to guess the proper type from the parameter types (Fallback logic required by the protobuf integration in Infinispan)".

That being said, it might be worth it to add a comment here?